web/src/components/contactForm.js

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import tw from 'twin.macro';
+
+const ContactWrapper = styled.div`
+    ${tw`text-center`}
+`;
+
+const FormStyling = styled.form`
+    ${tw`flex flex-col justify-center items-center`}
+`;
+
+const StyledHeading = styled.h1`
+    ${tw`font-bold mb-8 relative text-4xl text-black-3 leading-tight`}
+`;
+const InputWrapper = styled.div`
+    ${tw`flex gap-1 w-1/3 flex-wrap`}
+`;
+
+const ButtonStyling = styled.button`
+    ${tw`cursor-pointer py-2 px-6 max-w-max mt-16 text-xl tracking-widest`}
+
+    background: #00A09B;
+    transition: background 500ms ease-out;
+
+    &:hover {
+        background: #027874;
+    }
+`;
+
+const InputStyle = styled.input`
+    ${tw`border-2`}
+`;
+
+export default function ContactForm() {
+    const [formData, setFormData] = useState(null);
+    const [submitForm, setSubmitForm] = useState(false);
+
+    const encode = (data) => {
+        return Object.keys(data)
+            .map(
+                (key) =>
+                    `${encodeURIComponent(key)}=${encodeURIComponent(
+                        data[key],
+                    )}`,
+            )
+            .join('&');
+    };
+
+    const handleChange = (event) => {
+        setFormData({ ...formData, [event.target.name]: event.target.value });
+    };
+
+    const handleSubmit = (e) => {
+        e.preventDefault();
+        const form = e.target;
+        fetch('/', {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+            body: encode({
+                'form-name': form.getAttribute('name'),
+                ...formData,
+            }),
+        })
+            .then(() => setSubmitForm(true))
+            .catch((error) => error);
+    };
+
+    return (
+        <ContactWrapper>
+            <StyledHeading>Request a Demo</StyledHeading>
+            {submitForm ? (
+                <StyledHeading>Thank you!</StyledHeading>
+            ) : (
+                <FormStyling
+                    method="post"
+                    action="thanks"
+                    onSubmit={handleSubmit}
+                    data-netlify="true"
+                    data-netlify-honeypot="bot-field"
```

Comment:
I'm not sure this will work for the honeypot field. Adding this `data` attribute doesn't add and input field to the form.

I think you need to add a hidden input field like is mentioned [here](https://docs.netlify.com/forms/spam-filters/)